File: Java/Arrays/Eliminacion_elegida.java
/*
Eliminacion en Java, Eliminacion de un elemento elegido de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 04/09/2026
*/
import java.util.Arrays;

public class Eliminacion_elegida {
    public static void main(String[] args){
        int[] input_arr = {11, 21, 31, 41, 51, 61};     //array estatico
        int index_del = 3;      //indice a elminar
        
        //imprimimos arreglo sin cambios
        System.out.println("Antes de la eliminacion, el array es: ");
        for (int elemento : input_arr){
            System.out.println(elemento);
        }

        //Eliminacion del primer elemento
        int[] new_arr = new int[input_arr.length - 1];

        System.arraycopy(input_arr, 0, new_arr, 0, index_del);
        System.arraycopy(input_arr, index_del + 1, new_arr, index_del, input_arr.length - index_del - 1);
        
        System.out.println("Despue de la eliminacion, el array es: " + Arrays.toString(new_arr));       //otra forma de imprimir pero devuelve un string
    }
}

File: C++/Arrays/Eliminacion_elegida.cpp
/*
Eliminacion en C++, Eliminacion de un elemento elegido de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 06/09/2026
*/

#include <iostream>
#include <vector>

using namespace std;
int main(){
    vector <int> input_arr = {1, 11, 21, 31, 41, 51};
    int tamano = sizeof(input_arr)/sizeof(input_arr[0]);
    int index_del {4};

    cout << "Antes de la Eliminacion, el array es: ";
    for (int elemento : input_arr){
        cout << elemento << " ";
    }

    //Eliminacion
    input_arr.erase(input_arr.begin() + index_del);     //elimina en el indice elegido

    cout << "\n\nDespues de la eliminacion, el array es: ";
    for (int elemento : input_arr){
        cout << elemento << " ";
    }

    return 0;
}